### ISSUE TITLE
Crashed on run through

### ISSUE THREAD

**Issue by jaredseale** [KeeyanGhoreshi/PokemonPlatinumSingleSequence#5]
I started a run and left it on in another room, so I wasn't paying super close attention to anything happening. I came back around 10-ish hours into the run to find the game still and this error in the lua console:

NLua.Exceptions.LuaScriptException: A .NET exception occured in user-code
System.ArgumentOutOfRangeException: saveslots are 1 through 10
Parameter name: slotNum
   at BizHawk.Client.Common.SaveStateApi.SaveSlot(Int32 slotNum, Boolean suppressOSD)

![image](https://github.com/user-attachments/assets/52a99ee6-0249-48c9-9469-51444e4b9c69)

I see you just pushed a change to do with save slots less than an hour ago, so I'll set everything up again and try it with the new code. Logging this issue just in case it happens again though!

**Comment by KeeyanGhoreshi** [comment]
Yes, the commit I just made fixes this issue.  Bizhawk version 2.9.1 crashes if it gets an incorrect saveslot num and there was a bug with the saveslot id.  Notably the version of bizhawk I use does not produce any error for this, so it slipped through :D

**Comment by jaredseale** [comment]
Good stuff! Started up another run to test it again.

**Comment by Morilli** [comment]
This issue is caused by `id` being a global variable instead of a local one. It gets overwritten with the window's x position here: https://github.com/KeeyanGhoreshi/PokemonPlatinumSingleSequence/blob/79c64dac10f14284f89bceeeb71a7f8d297a250b/commands/movement.lua#L660
I'm not sure why there is a `client.xpos()` check in the first place, but this could be fixed by simply adding `local id;` before https://github.com/KeeyanGhoreshi/PokemonPlatinumSingleSequence/blob/5c5f0c0be931ad0d2aafe277154fcb24fd3e672b/commands/1.lua#L1403-L1406

**Comment by ZacharyGiguere** [comment]
Yeah got this exact same issue and came to the same conclusion as Morilli. Now everything should be fixe. Also, you should try make your savestates working if they don't already. you can find help here https://github.com/KeeyanGhoreshi/PokemonPlatinumSingleSequence/issues/3#issuecomment-2440446833

**Comment by jasem8te** [comment]
I didn't get any lua errors but i did get a desync, around the 9:10 mark, and ended up wandering this town jumping in and out of the menu. pretty sure i watched him go out of this centre, beat the gym, go somewhere else briefly and came back to the town, then it seemed to desync. For a while the inputs were like 'r, u, l, d, r, u, l, r, u, l, d...' including a few inputs missing, not completing a full circle. He also seemed like he was supposed to be going through long grass, stopping on every step and entering battle. No screenshots sorry, unfortunately I restarted my run before I got to this thread.

**Comment by KeeyanGhoreshi** [comment]
It's difficult to pin down what could've went wrong, if you have the savestates that would be the most helpful.
